static/js/factories/posts_test.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+import { assert } from 'chai';
+import R from 'ramda';
+
+import {
+  makePost,
+  makeChannelPostList
+} from './posts';
+
+describe('posts factories', () => {
+  describe('makePost', () => {
+    it('should make a text post', () => {
+      let post = makePost();
+      assert.isString(post.id);
+      assert.isString(post.title);
+      assert.isNumber(post.upvotes);
+      assert.equal(0, post.downvotes);
+      assert.isBoolean(post.upvoted);
+      assert.isBoolean(post.downvoted);
+      assert.isString(post.author);
+      assert.isString(post.text);
+      assert.isNull(post.url);
+    });
+
+    it('should make a URL post', () => {
+      let post = makePost(true);
+      assert.isNull(post.text);
+      assert.isString(post.url);
+    });
+
+    it('should randomly generate username', () => {
+      let firstPost = makePost();
+      let secondPost = makePost();
+      assert.notEqual(firstPost.author, secondPost.author);
+    });
+
+    it('should randomly generate upvotes', () => {
+      let upvoteScores = R.range(1,20)
+        .map(makePost)
+        .map(R.prop('upvotes'));
+      assert.isAbove(new Set(upvoteScores).size, 1);
+    });
+
+    it('should randomly generate id', () => {
+      let firstPost = makePost();
+      let secondPost = makePost();
+      assert.notEqual(firstPost.id, secondPost.id);
```

Comment:
My concern more is that it could fail at an inopportune time (e.g. merge commit in `release` branch). We've been bit a few times by this type of test.